src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -32,7 +32,24 @@ internal static DacDeployOptions CreateSchemaCompareOptions(DeploymentOptions de
                 var prop = dacOptions.GetType().GetProperty(deployOptionsProp.Name);
                 if (prop != null)
                 {
-                    prop.SetValue(dacOptions, deployOptionsProp.GetValue(deploymentOptions));
+                    var optionVal = deployOptionsProp.GetValue(deploymentOptions);
+                    if (deployOptionsProp.Name == "DoNotDropObjectTypes" || deployOptionsProp.Name == "ExcludeObjectTypes")
+                    {
+                        prop.SetValue(dacOptions, optionVal);
+                    }
+                    else
+                    {
+                        // Get value of the DeploymentOptionsProps option value property
+                        var value = ((DeploymentOptionProps)optionVal).value;
+                        
+                        // Convert Int64 to Int32
```

Comment:
Why? Comments should strive to explain why something is being done, we can already see the code is already converting from Int64 to Int32. 

Is overflow not something you're worried about?